mach_nix/data/providers.py

```diff
@@ -256,20 +233,18 @@ def __init__(
         self.sdist_provider = sdist_provider
 
     def get_pkg_reqs(self, c: Candidate) -> Tuple[Optional[List[Requirement]], Optional[List[Requirement]]]:
-        if not self.nixpkgs.exists(c.name, c.ver):
-            raise Exception(f"Cannot find {c.name}:{c.ver} in nixpkgs")
         requirements = self.nixpkgs.get_requirements(c.name, c.ver)
         if requirements is not None:
             return list(parse_reqs(requirements)), None
-        install_reqs, setup_reqs = [], []
-        try:
-            return self.sdist_provider.get_pkg_reqs(c)
-        except PackageNotFound:
-            try:
-                # wheel provider only knows install deps
-                return self.wheel_provider.get_pkg_reqs(c)[0], None
-            except:
-                return None, None
+
+        candidates = [
+            candidate
+            for candidate in self.sdist_provider.all_candidates(c.name)
+            if candidate.ver == c.ver
+        ]
+        if len(candidates) > 0:
+            return self.sdist_provider.get_pkg_reqs(candidates[0])
+        return [], []
```

Comment:
Any reason why you removed the fallback of getting the intall_requirements from the wheel provider?

Reply:
No. It looks like I didn't read the diff from https://github.com/DavHau/mach-nix/commit/b707c72fff14acbac936827cc4e9c6dcafa7dfd0 clearly enough when resolving conflicts.

Reply:
Fixed.